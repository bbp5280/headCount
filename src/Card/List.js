import React from 'react';
import styles from '../index.css';
import PropTypes from 'prop-types';

const List = ({data}) => {
  const yearArray = Object.keys(data)


  return (
    <div>
      {
        yearArray.map(year => {
        return (
          <div className="info-container" key={year}>
            <span className="data">{data[year]}</span>
            <span className="year">{year}</span>
          </div>
        )
        })
        }
    </div>
  )
};


export default List;
