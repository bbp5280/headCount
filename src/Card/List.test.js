import React from 'react';
import List from './List';
import { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import config from '../../test/setup.js';

describe('LIST', () => {
  let wrapper = shallow(<List />);

  it('should ...', () => {

  });

});
