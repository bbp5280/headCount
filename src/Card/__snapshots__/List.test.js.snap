exports[`LIST should match the List snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <List
    data={
        Object {
            "2004": 0.24,
          }
    } />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section
          className="info-container">
          <span
                    className="data">
                    0.24
          </span>
          <span
                    className="year">
                    2004
          </span>
</section>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "2004",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="data">
              0.24
</span>,
            <span
              className="year">
              2004
</span>,
          ],
          "className": "info-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": 0.24,
              "className": "data",
            },
            "ref": null,
            "rendered": 0.24,
            "type": "span",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "2004",
              "className": "year",
            },
            "ref": null,
            "rendered": "2004",
            "type": "span",
          },
        ],
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="info-container">
            <span
                        className="data">
                        0.24
            </span>
            <span
                        className="year">
                        2004
            </span>
</section>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "2004",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="data">
                0.24
</span>,
              <span
                className="year">
                2004
</span>,
            ],
            "className": "info-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": 0.24,
                "className": "data",
              },
              "ref": null,
              "rendered": 0.24,
              "type": "span",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "2004",
                "className": "year",
              },
              "ref": null,
              "rendered": "2004",
              "type": "span",
            },
          ],
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
